toolsrc/src/vcpkg/binarycaching.cpp

```diff
@@ -244,12 +245,541 @@ namespace
+            text(content);
+            Strings::append(buf, '"');
+            return *this;
+        }
+        XmlSerializer& finish_complex_open_tag()
+        {
+            Strings::append(buf, '>');
+            return *this;
+        }
+        XmlSerializer& finish_self_closing_complex_tag()
+        {
+            Strings::append(buf, "/>");
+            indent -= 2;
+            return *this;
+        }
+        XmlSerializer& close_tag(StringLiteral sl)
+        {
+            Strings::append(buf, "</", sl, '>');
+            indent -= 2;
+            return *this;
+        }
+        XmlSerializer& text(StringView sv)
+        {
+            Unicode::Utf8Decoder decoder(sv.begin(), sv.end());
+            for (auto ch : decoder)
+            {
+                if (ch == '&')
+                {
+                    buf.append("&amp;");
+                }
+                else if (ch == '<')
+                {
+                    buf.append("&lt;");
+                }
+                else if (ch == '>')
+                {
+                    buf.append("&gt;");
+                }
+                else if (ch == '"')
+                {
+                    buf.append("&quot;");
+                }
+                else if (ch == '\'')
+                {
+                    buf.append("&apos;");
+                }
+                else
+                {
+                    Unicode::utf8_append_code_point(buf, ch);
+                }
+            }
+            return *this;
+        }
+        XmlSerializer& simple_tag(StringLiteral tag, StringView content)
+        {
+            return open_tag(tag).text(content).close_tag(tag);
+        }
+        XmlSerializer& line_break()
+        {
+            buf.push_back('\n');
+            buf.append(indent, ' ');
+            return *this;
+        }
+    };
+
+    struct NugetReference
+    {
+        explicit NugetReference(const Dependencies::InstallPlanAction& action)
+            : NugetReference(action.spec,
+                             action.source_control_file_location.value_or_exit(VCPKG_LINE_INFO)
+                                 .source_control_file->core_paragraph->version,
+                             action.abi_info.value_or_exit(VCPKG_LINE_INFO).package_abi)
+        {
+        }
+
+        NugetReference(const PackageSpec& spec, const std::string& raw_version, const std::string& abi_tag)
+            : id(spec.dir()), version(reformat_version(raw_version, abi_tag))
+        {
+        }
+
+        std::string id;
+        std::string version;
+
+        std::string nupkg_filename() const { return Strings::concat(id, '.', version, ".nupkg"); }
+
+    private:
+        static std::string trim_zeroes(std::string v)
+        {
+            auto n = v.find_first_not_of('0');
+            if (n == std::string::npos)
+            {
+                return "0";
+            }
+            if (n > 0)
+            {
+                v.erase(0, n);
+            }

```

Comment:
```suggestion
            if (n == std::string::npos)
            {
                v = "0";
            }
            else if (n > 0)
            {
                v.erase(0, n);
            }

```

Hopefully always returning the same object is kinder to the optimizer (and I think it is kinder to humans too but nitpick here)